packages/react-components/react-input/src/components/Input/useInputStyles.styles.ts

```diff
@@ -288,26 +345,30 @@ export const useInputStyles_unstable = (state: InputState): InputState => {
   state.input.className = mergeClasses(
     inputClassNames.input,
     useInputClassName(),
-    size === 'large' && inputStyles.large,
+    inputStyles[size],
```

Comment:
Add the inputHasBeforeClassName, etc here instead of below inside the `if (state.contentBefore)`. The way you're adding them below will cause them to be overridden by `inputStyles[size]` since they'll be added ahead of those classes (later classes override earlier ones with atomic css).
```suggestion
    inputStyles[size],
    state.contentBefore && inputHasBeforeClassName[size],
    state.contentAfter && inputHasAfterClassName[size],
```